extensions/control-plane/store/cosmos/contract-definition-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStoreIntegrationTest.java

```diff
@@ -337,14 +342,34 @@ void verify_readWriteFindAll() {
 
     }
 
+    @Test
+    @Disabled("Cosmos does not support yet matching a JSON")
+    void find_queryBySelectorExpression_entireCriterion() throws JsonProcessingException {
+        var definitionsExpected = createContractDefinitions(20);
+        definitionsExpected.get(0).getSelectorExpression().getCriteria().add(new Criterion(Asset.PROPERTY_ID, "=", "test-asset"));
+        var def5 = definitionsExpected.get(5);
+        def5.getSelectorExpression().getCriteria().add(new Criterion(Asset.PROPERTY_ID, "=", "foobar-asset"));
+        getContractDefinitionStore().save(definitionsExpected);
+
+        var json = new ObjectMapper().writeValueAsString(new Criterion(Asset.PROPERTY_ID, "=", "foobar-asset"));
+
+        var spec = QuerySpec.Builder.newInstance()
+                .filter("selectorExpression.criteria = " + json)
+                .build();
+
+        assertThat(getContractDefinitionStore().findAll(spec)).hasSize(1)
+                .usingRecursiveFieldByFieldElementComparator()
+                .containsOnly(def5);
+    }
+
```

Comment:
I would add the following test: 
```java
    @Test
    void find_queryBySelectorExpression_rightAndLeft() {
        var definitionsExpected = createContractDefinitions(20);
        definitionsExpected.get(0).getSelectorExpression().getCriteria().add(new Criterion(Asset.PROPERTY_ID, "=", "test-asset"));
        definitionsExpected.get(5).getSelectorExpression().getCriteria().add(new Criterion(Asset.PROPERTY_ID, "=", "foobar-asset"));
        getContractDefinitionStore().save(definitionsExpected);

        var spec = QuerySpec.Builder.newInstance()
                .filter(List.of(
                        new Criterion("selectorExpression.criteria.operandLeft", "=", Asset.PROPERTY_ID),
                        new Criterion("selectorExpression.criteria.operandRight", "=", "foobar-asset")))
                .build();

        var definitionsRetrieved = getContractDefinitionStore().findAll(spec).collect(Collectors.toList());

        assertThat(definitionsRetrieved).hasSize(1)
                .usingRecursiveFieldByFieldElementComparator()
                .containsOnly(definitionsExpected.get(5));
    }
```
I've tried it out, it works, but we should guard against regression, so we can make sure that concatenating two `Criterion`s also works fine.